src/libtls/tests/suites/test_socket.c

```diff
@@ -442,6 +442,63 @@ static void test_tls(tls_version_t version, uint16_t port, bool cauth, u_int i)
 	free(config);
 }
 
+/**
+ * Diffie-Hellman group names as they can be used in config files.
+ */
+ENUM_BEGIN(dh_group_config_names, ECP_256_BIT, ECP_521_BIT,
+	"ecp256",
+	"ecp384",
+	"ecp521");
+ENUM_NEXT(dh_group_config_names, ECP_192_BIT, ECP_224_BIT, ECP_521_BIT,
+	"ecp192",
+	"ecp224");
+ENUM_NEXT(dh_group_config_names, CURVE_25519, CURVE_448, ECP_224_BIT,
+	"curve25519",
+	"curve448");
+ENUM_END(dh_group_config_names, CURVE_448);
+
+/**
+ * TLS curve test wrapper function
+ */
+static void test_tls_curves(tls_version_t version, uint16_t port, bool cauth,
+							u_int i)
+{
+	echo_server_config_t *config;
+	diffie_hellman_group_t *curves;
+	char curve[128];
+	int count;
+
+	INIT(config,
+		.version = version,
+		.addr = "127.0.0.1",
+		.port = port,
+		.cauth = cauth,
+		.data = chunk_from_chars(0x01,0x02,0x03,0x04,0x05,0x06,0x07,0x08),
+	);
+
+	start_echo_server(config);
+
+	count = tls_crypto_get_supported_curves(&curves);
+	ck_assert(i < count);
+	snprintf(curve, sizeof(curve), "%N", dh_group_config_names, curves[i]);
```

Comment:
Maybe there is a better more generic way to map `diffie_hellman_group_names` such as `ECP_256` to `ecp256` as `get_token()` expects it. I noted also that the names has to be in lower case.

Reply:
Yes `--ignore-case` is required.